### ISSUE TITLE
Nested columns not visible when connecting to prestoDB

### ISSUE THREAD

**Issue by josephfrancis** [apache/incubator-superset#3243]
Make sure these boxes are checked before submitting your issue - thank you!

- [x ] I have checked the superset logs for python stacktraces and included it here as text if any
- [x ] I have reproduced the issue with at least the latest released version of superset
- [x ] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.19.0

### Expected results
To be able to see the nested columns in SQL LAB (on the left pane) and to have autosuggest pick up the nested column names in the query editor. 
If there is a nested column called header.date.epoch_time, I only see the header column.
Note: I'm connecting to hive through prestoDB

### Actual results
I can only see the top level columns. E.g: header, whereas it will be good to be able to drill down and see the nested column name; header.date.epoch_time.

### Steps to reproduce
Connect to hive through prestoDB.
Create a hive table from a nested parquet file.

**Comment by mistercrunch** [comment]
Is that info even in the metastore? Any idea what the API to discover that would be?

**Comment by josephfrancis** [comment]
Yeah the info is in the metastore, as we can see the column breakdown when seeing the same table through hue. `SHOW COLUMNS FROM table` query in presto lists all the nested column data types as well.
Not quite sure yet whether its currently returned to superset though.

**Comment by josephfrancis** [comment]
I take it superset uses pyhive for connection to presto as well? If thats the case it's maybe not returned by pyhive: https://github.com/dropbox/PyHive/blob/master/pyhive/sqlalchemy_presto.py#L38
The `_type_map` there doesn't seem to list any of the complex/nested row types.

**Comment by mistercrunch** [comment]
Yes we use pyhive. We're committed to allow to go around the lowest common denominator issue with SQLAlchemy. We typically put the specifics in the `superset.db_engine_specs` module. For instance we could add a custom `get_table_columns` method in there that would default to the current SQLAlchemy logic in BaseEngineSpecs, but that would implement a solution that involves `SHOW COLUMNS FROM table` in the case of Presto.

It may not make the top of Airbnb's priority list for a little while, but we'll happily review a PR that adds functionality in that direction.
